packages/esm-patient-chart-app/src/visit/visit-action-items/delete-visit-action-item.component.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { Button } from '@carbon/react';
+import { UserHasAccess, Visit, showModal, useLayoutType } from '@openmrs/esm-framework';
+import { useTranslation } from 'react-i18next';
+import { TrashCan } from '@carbon/react/icons';
+
+interface DeleteVisitActionItemProps {
+  patientUuid: string;
+  visit: Visit;
+}
+
+const DeleteVisitActionItem: React.FC<DeleteVisitActionItemProps> = ({ patientUuid, visit }) => {
+  const { t } = useTranslation();
+  const isTablet = useLayoutType() === 'tablet';
+
+  const deleteVisit = () => {
+    const dispose = showModal('delete-visit-dialog', {
+      patientUuid,
+      visit,
+      closeModal: () => dispose(),
+    });
+  };
+
+  const cancelVisit = () => {
+    const dispose = showModal('cancel-visit-dialog', {
+      patientUuid,
+      closeModal: () => dispose(),
+    });
+  };
+
+  const isActiveVisit = !visit?.stopDatetime;
+
+  if (visit?.encounters?.length) {
+    return null;
+  }
+
+  return (
+    <UserHasAccess privilege="Delete Visits">
+      <Button
+        onClick={isActiveVisit ? cancelVisit : deleteVisit}
+        kind="danger--ghost"
+        renderIcon={TrashCan}
+        size={isTablet ? 'lg' : 'sm'}
+      >
+        {isActiveVisit ? t('cancelVisit', 'Cancel visit') : t('deleteVisit', 'Delete')}
+      </Button>
+    </UserHasAccess>
+  );
```

Comment:
> I don't see any checks to ensure that a visit is empty when it's deleted. In fact, there's logic in here to cascade visit deletes to also delete visit queue entries.

Hi @ibacher , the delete button is only rendered if the visit is empty, otherwise it isn't rendered.

Reply:
@vasharma05 But the check here for `isActiveVisit` is `!visit?.stopDatetime`. I don't see how that equates to "the visit is empty"? That just equates to "the visit is empty". Basically, this seems to render the "Delete" button (and call `deleteVisit`) if the visit has a `stopDateTime`. What am I missing?

Reply:
Please see the first 2 lines
```
if (visit?.encounters?.length) {
  return null;
}
```